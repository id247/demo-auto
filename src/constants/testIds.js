export default {
  button: 'button',
  select: 'select',
  selectButton: 'select-button',
  input: 'input'
};
