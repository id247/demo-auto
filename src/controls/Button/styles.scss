@import "styles/variables.scss";

.button {
  border: 0;
  border-radius: $radius-s;
  padding: $space-s $space-s;
  font-size: $font-size-m;
  width: 128px;
  height: 32px;
  cursor: pointer;
  transition: background-color 0.2s;

  &:hover {
    background-color: lighten($color-orange, 10%);
  }

  &:active {
    background-color: $color-brown;
  }

  &--primary {
    background-color: $color-orange;
    color: #fff;
  }
}
