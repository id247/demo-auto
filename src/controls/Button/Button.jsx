import React from 'react';
import PropTypes from 'prop-types';
import testIds from 'constants/testIds';
import './styles.scss';

function Button({ type = 'button', children, ...props }) {
  return (
    <button
      data-testid={testIds.button}
      type={type}
      className="button button--primary"
      {...props}
    >
      {children}
    </button>
  );
}

Button.propTypes = {
  children: PropTypes.node,
  type: PropTypes.string
};

export default Button;
