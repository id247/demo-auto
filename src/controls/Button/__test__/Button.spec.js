import 'jest-dom/extend-expect';
import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import testIds from 'constants/testIds';
import Button from '../';

afterEach(cleanup);

describe('Button component', () => {
  test('Renders with defaults', async () => {
    const { container, getByTestId } = render(<Button />);

    const button = getByTestId(testIds.button);
    expect(button).toHaveClass('button button--primary');
    expect(button).toHaveAttribute('type', 'button');
    expect(container.firstChild).toMatchSnapshot();
  });

  test('Renders with props', async () => {
    const text = 'Button';
    const type = 'submit';
    const { container, getByTestId } = render(
      <Button type={type}>{text}</Button>
    );

    const button = getByTestId(testIds.button);
    expect(button).toHaveTextContent(text);
    expect(button).toHaveClass('button button--primary');
    expect(button).toHaveAttribute('type', type);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('Handles events', async () => {
    const onClick = jest.fn();
    const { getByTestId } = render(<Button onClick={onClick} />);

    const button = getByTestId(testIds.button);
    fireEvent.click(button);
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
