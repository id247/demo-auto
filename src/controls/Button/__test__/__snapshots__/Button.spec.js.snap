// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Renders with defaults 1`] = `
<button
  class="button button--primary"
  data-testid="button"
  type="button"
/>
`;

exports[`Button component Renders with props 1`] = `
<button
  class="button button--primary"
  data-testid="button"
  type="submit"
>
  Button
</button>
`;
